### PR TITLE
Update lintrunner to version that uses  as default mergebase

### LINKED ISSUE #93156: Local and CI lintrunner should be more consistent
At least twice folks have run into situations where the CI linter would give an error that they didn't think was related to their changes, wouldn't show up when they ran lintrunner locally, and so they force merged assuming the failure wasn't their fault.

**Instance one:**
Dev couldn't repo the lint errors on his [PR](https://github.com/pytorch/pytorch/pull/92637)

Probably this was the set of [confusing linter errors](https://github.com/pytorch/pytorch/actions/runs/3963249122/jobs/6790866996). If not, it would be something from [this list](https://github.com/pytorch/pytorch/actions/workflows/lint.yml?query=branch%3Avoz%2Ftiming+workflow%3ALint)

This seems to have been missed by the local linter since the local one only looks at changes that go one commit back

**Instance two:**
Don't have the PR for this right now, but Quantsight reported that they took a dependency on some Dynamo/Inductor package which introduced a linter error. The error occurred since Inductor/Dynamo packages are currently excluded from some linting/policy enforcement rules but any other package that imports from them has to follow those rules, and those requirements then transversely become imposed on the inductor/dynamo packages.

However, the local lintrunner doesn't apply those transverse errors, making the local lintrunner say things are fine while the CI linter fails.

### PR DESCRIPTION
**Pull request by ZainRizvi**
Fixes https://github.com/pytorch/pytorch/issues/93156

Upgrades lintrunner to the latest version which can use the `master` branch as the merge base by default (provided it's specified in `.lintrunner.toml` and update `.lintrunner.toml` accordingly

### CODE REVIEW

**Review by huydhn — APPROVED** [review]
What's happened with everyone from 0.9.3 to 0.10.6? :P

**Comment by ZainRizvi** [comment]
> What's happened with everyone from 0.9.3 to 0.10.6? :P

https://github.com/suo/lintrunner/blob/main/CHANGELOG.md has all the deets :)

File: .github/workflows/_calculate-docker-image.yml [hunk 1]
```diff
@@ -25,7 +25,8 @@ jobs:
         uses: pytorch/pytorch/.github/actions/checkout-pytorch@master
         with:
           submodules: false
-          fetch-depth: 1
+          # A very high bound which should be more than we ever need.
+          fetch-depth: 200
```
Comment (huydhn) [inline .github/workflows/_calculate-docker-image.yml:29]: Hmm, I'm trying to understand why this is needed here. This step fails if the Docker image has been built yet.  But once the image is ready, rerunning it works (that's what works for me)

**Comment by huydhn** [comment]
> @huy fyi, I had to update the fetch depth in `.github/workflows/_calculate-docker-image.yml`, since otherwise it was impossible for the `calculate-docker-image` action to find a merge base.
> 
> Seems like we hadn't run into this corner case in the past three weeks since docker images were added for linters

Oh I see.  I'm about to ask that question.  Here is what works for me FYI: This step fails if the Docker image has been built yet. But once the image is ready, rerunning it works.

**Pull request closed without merging**